stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/decorator/internal/FootnoteDecorator.kt

```diff
@@ -146,8 +150,9 @@ internal class FootnoteDecorator(
                 textView.isVisible = true
                 style.textStyleUserName.apply(textView)
             }
-            data.isNotBottomPosition() -> textView.isVisible = false
-            !data.message.isEphemeral() && !data.message.isDeleted() -> textView.isVisible = false
+            data.isNotBottomPosition()
+                || !data.message.isEphemeral() && !data.message.isDeleted()
+                || deletedMessageListItemPredicate != VisibleToAuthorOnly -> textView.isVisible = false
```

Comment:
🤔 Wouldn't it break ephemeral messages? I think this condition should be applied only when `data.message.isDeleted()`

Reply:
The boolean logic looks the same to me, I just put then in the same place. If we apply the condition only then the message isDeleted, we would be changing the boolean logic, no?

Reply:
🤔 Imagine `message.isEphemeral() == true` and `message.isDeleted() == false`.
Then we have: `!data.message.isEphemeral() && !data.message.isDeleted()` == `false && true` == `false` and we go to the third condition `deletedMessageListItemPredicate != VisibleToAuthorOnly` which might be false.
This `when` looks very complicated overall

Reply:
This example looks correct to me, we would like to hide the textView if any of those conditions are true. Would you prefer this way? 

```
data.isNotBottomPosition() -> textView.isVisible = false

!data.message.isEphemeral() && !data.message.isDeleted() -> textView.isVisible = false

deletedMessageListItemPredicate != VisibleToAuthorOnly -> textView.isVisible = false
```

The boolean logic is the same, but it looks closer to the original implementation.

Reply:
Or when we can change it to use positive check, instead of negative ones:
```
 when {
    data.isBottomPosition() && !isDirectMessage() && data.isTheirs -> {
        textView.text = data.message.user.name
        textView.isVisible = true
        style.textStyleUserName.apply(textView)
    }

    data.isBottomPosition()
        && data.message.isEphemeral()
        && data.message.isDeleted()
        && deletedMessageListItemPredicate == VisibleToAuthorOnly -> {
            textView.run {
                isVisible = true
                text = context.getString(R.string.stream_ui_message_list_ephemeral_message)
                setLeftDrawable(style.iconOnlyVisibleToYou)
                compoundDrawablePadding = resources.getDimensionPixelSize(R.dimen.stream_ui_spacing_small)
            }
        }
}
```

Reply:
I like the second version. Maybe we can simplify it to:
``` 
when {
    data.isBottomPosition() && !isDirectMessage() && data.isTheirs -> {
        textView.text = data.message.user.name
        textView.isVisible = true
        style.textStyleUserName.apply(textView)
    }

    data.isBottomPosition()
        && data.message.isDeleted()
        && deletedMessageListItemPredicate == VisibleToAuthorOnly -> {
        showOnlyVisibleToYou(textView)
    }

    data.isBottomPosition() && data.message.isEphemeral() -> {
        showOnlyVisibleToYou(textView)
    }
  else -> {
//hide
}
}
```